corehq/apps/locations/models.py

```diff
@@ -242,10 +244,33 @@ class SQLLocation(MPTTModel):
     # This should really be the default location manager
     active_objects = OnlyUnarchivedLocationManager()
 
+    @classmethod
+    def _migration_get_fields(cls):
+        return ["domain", "name", "lineage", "site_code", "external_id",
+                "metadata", "is_archived", "latitude", "longitude"]
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return Location
+
+    def _migration_do_sync(self):
+        couch_obj = self._migration_get_or_create_couch_object()
+        couch_obj._sql_location_type = self.location_type
+        self._migration_sync_to_couch(couch_obj)
+
     def save(self, *args, **kwargs):
         from corehq.apps.commtrack.models import sync_supply_point
         self.supply_point_id = sync_supply_point(self)
-        return super(SQLLocation, self).save(*args, **kwargs)
+
+        sync_to_couch = kwargs.pop('sync_to_couch', True)
+        kwargs['sync_to_couch'] = False  # call it here
+        super(SQLLocation, self).save(*args, **kwargs)
```

Comment:
Looks like it was sort of an accident that this works.  I think the answer is [here](https://docs.djangoproject.com/en/1.9/ref/models/fields/#null):
> Avoid using null on string-based fields such as CharField and TextField because empty string values will always be stored as empty strings, not as NULL.

So `SQLLocation.objects.create(<don't set location_id>)` creates a location with `location_id == ""`.